server/realTime/scripts/Network.js

```diff
@@ -116,7 +126,15 @@ Protocol.prototype.updateProtectedRegions = function updateProtectedRegions (roo
 				new SAT.Vector(data[k].maxX, data[k].maxY),
 				new SAT.Vector(data[k].minX, data[k].maxY),
 			]);
-
+			for(var f = minRegionId; f < permissions.length; f++){
+				if(permissions[f].regionId == data[k].id){
+					if(typeof data[k].permissions === "undefined")
+						data[k].permissions = new Array();
```

Comment:
You also should just write this as data[k].permissions = data[k].permissions || [] instead of using an if